rxgroups-android/src/main/java/com/airbnb/rxgroups/GroupLifecycleManager.java

```diff
@@ -84,19 +78,14 @@ public static GroupLifecycleManager onCreate(ObservableManager observableManager
 
       Preconditions.checkState(state != null, "Must call onSaveInstanceState() first");
 
-      // First check the instance hashCode before restoring state. If it's not the same instance,
+      // First check the instance ID before restoring state. If it's not the same instance,
       // then we have to create a new group since the previous one is already destroyed.
       // Android can sometimes reuse the same instance after saving state and we can't reliably
```

Comment:
`Android can sometimes reuse the same instance after saving state` I don't think this comment is so accurate any more. I think the issue is that groups are cleared when a new observable manager instance is used (usually because of the process restarting)